### ISSUE TITLE
Default handler icon?

### ISSUE THREAD

**Issue by zkoch** [w3c/webpayments-payment-apps-api#178]
I don't see a place in the spec defined for how one finds the default icon for a given handler. I suspect that strong brands will want their primary icon + name displayed. Did I miss something?

**Comment by rsolomakhin** [comment]
We can achieve this through a web app manifest on the page that registered the service worker. This should be in the spec, IMHO.

**Comment by ianbjacobs** [comment]
I agree the specification is lacking here. My understanding from our previous discussions is that:

 * As Rouslan indicates, we expected web apps to use web app manifest for app-level information.
   This is hinted at in the spec in the sentence "The user agent displays and groups these choices 
    according to information (labels and icons) provided at registration or otherwise available 
    from the Web app." The hint is not that helpful.
 * The instrument-level icons and names are not addressed by web app manifest and thus
    we define fields in this API.

It seems we should add a sentence like the following to the spec (e.g., at the beginning of section 6 after the first paragraph), such as:
<blockquote>
This API defines mechanisms to specify icons and labels for the display of instruments. Other Web technology (e.g., Web app manifest) enables user agents to fetch icons, labels, and other "higher level" information about the payment handler.</p>
</blockquote>

Ian

**Comment by rsolomakhin** [comment]
I propose to recommend a web app manifest or its `<link>` equivalent fallback. If no labels or icons are found after all of the fallbacks, a browser can degrade the experience gracefully.

For example, an unlabeled payment handler can be sorted to the bottom of the list of payment apps and be identified only by its origin:

1. [Label] [Origin] [Icon]
1. [Label] [Origin]
1. [Origin] [Icon]
1. [Origin]

It would be prudent for the unlabeled payment handler to print a developer console warning, "No label/icon found for payment handler. Experience is degraded."

**Comment by ianbjacobs** [comment]
Thank you, @rsolomakhin, for writing this up following the 20 June payment apps call [1]. Here were the four take-away points (mostly covered by your post):

* We include a statement in the spec that says that app-level information is provided through other web technologies (e.g., web app manifest and fallback approaches).
* When there are no icons and labels available (for any of a number of reasons), we leave to the browser how to best manage a fallback user experience (e.g., using origin as you indicate; we also
mentioned favicons). 
* Browsers should not omit a matching payment handler from the list merely because of missing icons and labels.

Also +1 to the console warning.

Ian



[1] https://www.w3.org/2017/06/20-apps-minutes.html#item05

**Comment by ianbjacobs** [comment]
See discussion at 22 August call:
 http://www.w3.org/2017/08/22-apps-minutes.html

Ian
